File: 함수/1065 한수(2).c
////////// C2 //////////
// 예전에 푼 방식
// 조건이 1000 '이하'인데 1000일 때를 딱히 고려하지 않은 듯...?
// 굳이 안 걸리는 부분이라서 그런가

#define _CRT_SECURE_NO_WARNINGS
#include<stdio.h>

int han(int n);

int main() {

	int input;

	scanf("%d", &input);
	han(input);
	
	return 0;
}

int han(int n) {
	int cnt = 0;
	int h, t, o;

	if (n < 100) printf("%d", n);
	else {
		for (int i = 100; i <= n; i++) {
			h = i / 100;
			t = (i / 10) % 10;
			o = i % 10;

			if ((h - t) == (t - o)) cnt++;
		}
		printf("%d", cnt + 99);
	}

}

File: 함수/1065 한수(2).py
########## Python 2 ##########
# 1번 좀 더 간단하게

a = int(input())
cnt = 0

for i in range(1, a+1):
    if i < 100:
        cnt += 1 # iteration 동안 여기를 += 1로 해서
    else:
				# 여기서 굳이 99에서 시작해서 for 새로 만들고 돌지 않아도 됨
        arr = list(map(int, str(i)))
        if arr[0] - arr[1] == arr[1] - arr[2]:
            cnt += 1

print(cnt)


# 68ms 걸림
